paymentsheet/src/test/java/com/stripe/android/customersheet/CustomerSheetViewModelTest.kt

```diff
@@ -0,0 +1,70 @@
+package com.stripe.android.customersheet
+
+import android.content.Context
+import androidx.test.core.app.ApplicationProvider
+import app.cash.turbine.test
+import com.google.common.truth.Truth.assertThat
+import com.stripe.android.ApiKeyFixtures
+import com.stripe.android.PaymentConfiguration
+import com.stripe.android.ui.core.forms.resources.LpmRepository
+import kotlinx.coroutines.test.runTest
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+@OptIn(ExperimentalCustomerSheetApi::class)
+class CustomerSheetViewModelTest {
+    private val context = ApplicationProvider.getApplicationContext<Context>()
+    private val lpmRepository = LpmRepository(
+        LpmRepository.LpmRepositoryArguments(
+            resources = context.resources,
+            isFinancialConnectionsAvailable = { true },
+            enableACHV2InDeferredFlow = true
+        )
+    )
+
+    @Before
+    fun setup() {
+        PaymentConfiguration.init(
+            context,
+            ApiKeyFixtures.FAKE_PUBLISHABLE_KEY
+        )
+    }
+
+    @Test
+    fun `ViewModel initializes with loading`() = runTest {
+        val viewModel = createViewModel()
+        viewModel.viewState.test {
+            assertThat(awaitItem()).isEqualTo(CustomerSheetViewState.Loading)
+        }
+    }
+
+    private fun createViewModel(
+        customerEphemeralKeyProvider: CustomerEphemeralKeyProvider =
+            CustomerEphemeralKeyProvider {
+                Result.success(
+                    CustomerEphemeralKey(
+                        customerId = "cus_123",
+                        ephemeralKey = "ek_123",
+                    )
+                )
+            },
+        setupIntentClientSecretProvider: SetupIntentClientSecretProvider =
+            SetupIntentClientSecretProvider {
+                Result.success("seti_123")
+            },
+        customerAdapter: CustomerAdapter = CustomerAdapter.create(
```

Comment:
We should remove the `customerAdapter` param, and create it inline (since we're already passing in the things needed to create the customer adapter).

Or we could do it the other way around (remove customerEphemeralKeyProvider/setupIntentClientSecretProvider) and require the caller to pass that in or use the default.